here/here_readme.md

```diff
@@ -0,0 +1,218 @@
+
+```sql
+JOIN here_gis.streets_att_16_1 gis ON gis.link_id = LEFT(ta.link_dir, -1)::numeric
+```
+
+#### Functional Class 5
+
+This bucket contains a little bit of everything that doesn't fall into the other classes. Currently exclude
+
+`"paved"  = 'Y' AND "poiaccess" =  'N' AND "ar_auto" = 'Y' AND "ar_traff" = 'Y'`
+
+## GIS Data
+
+A lot of map layers provided by HERE, see the [README](gis/README.md) in the [gis](gis/) folder for more info.
+
+## Traffic Patterns: Traffic Models
+
+Just like the sun doesn't always shine, the streets of Toronto don't always produce vehicle probe speeds. In those cases, HERE provides us with traffic patterns, a model for each street link by time of week.
+
+### Traffic Patterns: Data Model 
+
+The Traffic Patterns dataset comes in 15-min and 60-min models, which have a basic wide structure of `{pattern_id, h00_00, h00_015, [...], h23_45}` where `hHH_MM` is the speed value for `pattern_id` for that time of day. 
+
+Both the 15-min and 60-min tables share the same `pattern_ids`, which can be found in the lookup reference table `here.traffic_pattern_YY_ref` (where `YY` is the year of the model). This table is of the format `{link_pvid, travel_direction, u, m, t, w, r, f, s}` where each of those letter columns contains a `pattern_id` for that combination of `link_dir` and `day of the week` starting with sUnday.
+
+`here.traffic_pattern_YY_ref_narrow`
+
+|column | type | definition |
+|-------|------|------------|
+|link_dir |text | link direction |
+|isodow |integer | ISO Day of Week|
+|pattern_id |integer | id referring to the pattern for that link_dir, day of week combination|
+
+The tables are converted to a narrow format (which means that the 15-min and 60-min time periods show up in one field (`trange`) instead of each time period taking up its own column). The 15-min and 60-min tables have the same format.
+
+`here.traffic_pattern_YY_spd_MM_narrow`
+
+|column | type | definition |
+|-------|------|------------|
+|pattern_id| integer| ID for this pattern|
+|trange| timerange| Time range for which this pattern applies|
+|pattern_speed| integer|Speed in km/hr for that pattern_id & time range|
+
+Check out this [documentation](here/here_loading.md) for more info on how the traffic patterns tables are loaded and transformed.
+
+## Routing with Traffic Data
+
+One use of historical traffic data is the ability to route a vehicle from an
+arbitrary point to another arbitrary point using traffic data **at that point
+in time**. Since our data are already in a database, this can be accomplished
+using the [`pgRouting`](http://pgrouting.org/) PostgreSQL extension. It is
+necessary to have [traffic patterns](#traffic-patterns-traffic-models) loaded
+to fill in gaps in traffic data in time.
+
+The following views prepare the HERE data for routing (code found
+[here](traffic/sql/create_here_routing.sql)):
+
+- `here.routing_nodes`: a view of all intersections derived from the `z_levels`
```

Comment:
this should also have a year & revision

Reply:
We get their latest map updates every year - then there's a day when the ta table starts being updated to the new map. 

That's nice and all, but there should be a suffix reference _yy_r (and then also update the at a glance table).